transactions/generic_destroy_nft.cdc

```diff
@@ -0,0 +1,86 @@
+/// This transaction withdraws one or more NFTs from the signers collection and destroys them
+/// The NFTs must all be of the same type
+
+/// Imports to replace with "from {Address}"
+import "NonFungibleToken"
+import "MetadataViews"
+import "Burner"
+
+/// @param contractAddress: The address of the contract for the NFTs you want to destroy
+///                         Example: For Top Shot on mainnet, 0x0b2a3299cc857e29
+/// @param contractName: The name of the contract for the NFTs you want to destroy.
+///                      Example: "TopShot", "AllDay", etc
+/// @param nftTypeName: The name of the NFT type you want to destroy.
+///                     99% of the time it is "NFT"
+/// @param ids: An array of ID of the NFTs you would like to destroy
+
+/// Replace each instance of the params above with the values that you want to use
+/// Example (With Top Shot):
+/// transaction(contractAddress: 0x0b2a3299cc857e29, contractName: "TopShot", nftTypeName: "NFT", ids: [14337, 337282, 3722711]) {
+
+transaction(contractAddress: Address, contractName: String, nftTypeName: String, ids: [UInt64]) {
+
+    /// NFTCollectionData struct to get paths from
+    let collectionData: MetadataViews.NFTCollectionData
+
+    /// Reference to the owner's collection to destroy NFTs from
+    let withdrawRef: auth(NonFungibleToken.Withdraw) &{NonFungibleToken.Collection}
+
+    prepare(signer: auth(BorrowValue) &Account) {
+        // Borrow a reference to the nft contract deployed to the passed account
+        let resolverRef = getAccount(contractAddress)
+            .contracts.borrow<&{NonFungibleToken}>(name: contractName)
+                ?? panic("Could not borrow NonFungibleToken reference to the contract. Make sure the provided contract name "
+                          .concat(contractName).concat(" and address ").concat(contractAddress.toString()).concat(" are correct!"))
+
+        // Use that reference to retrieve the NFTCollectionData view 
+        self.collectionData = resolverRef.resolveContractView(resourceType: nil, viewType: Type<MetadataViews.NFTCollectionData>()) as! MetadataViews.NFTCollectionData?
```

Comment:
shouldn't `resourceType: nil` be `CompositeType(typeString)` ?

Reply:
yes, good catch